src/backend/graphics/opengl.rs

```diff
@@ -0,0 +1,17 @@
+use nix::c_void;
+
+pub enum SwapBuffersError {
+    ContextLost,
+    AlreadySwapped,
+}
+
+pub trait OpenGLRenderer
+{
+    fn swap_buffers(&self) -> Result<(), SwapBuffersError>;
+    unsafe fn get_proc_address(&self, symbol: &str) -> *const c_void;
+    fn get_framebuffer_dimensions(&self) -> (u32, u32);
+    fn is_current(&self) -> bool;
+    unsafe fn make_current(&self);
+    fn get_api(&self) -> Api;
+    fn get_pixel_format(&self) -> PixelFormat;
```

Comment:
Isn't this part of glutin/glium a direct expression of how OpenGL works?

I mean, for example I expect all graphics backends (glutin/x11/wayland/native) to be directly usable with OpenGL wrappers such as glium and gfx...

So I guess my question here is: what in this trait is glutin/glium-specific?